app/src/lib/git/checkout.ts

```diff
@@ -110,6 +117,72 @@ export async function checkoutBranch(
   return true
 }
 
+/**
+ * Check out the given commit.
+ * Literally invokes `git checkout <commit SHA>`.
+ *
+ * @param repository - The repository in which the branch checkout should
+ *                     take place
+ *
+ * @param commit     - The commit that should be checked out
+ *
+ * @param progressCallback - An optional function which will be invoked
+ *                           with information about the current progress
+ *                           of the checkout operation. When provided this
+ *                           enables the '--progress' command line flag for
+ *                           'git checkout'.
+ */
+export async function checkoutCommit(
+  repository: Repository,
+  account: IGitAccount | null,
+  commit: CommitOneLine,
+  progressCallback?: ProgressCallback
+): Promise<true> {
+  let opts: IGitExecutionOptions = {
+    env: await envForRemoteOperation(
+      account,
+      getFallbackUrlForProxyResolve(account, repository)
+    ),
+    expectedErrors: AuthenticationErrors,
+  }
+
+  if (progressCallback) {
+    const title = `Checking out commit ${commit.sha}`
+    const kind = 'checkout'
+    const targetCommit = commit.sha
+
+    opts = await executionOptionsWithProgress(
+      { ...opts, trackLFSProgress: true },
+      new CheckoutProgressParser(),
+      progress => {
+        if (progress.kind === 'progress') {
+          const description = progress.details.text
+          const value = progress.percent
+
+          progressCallback({
+            kind,
+            title,
+            description,
+            value,
+            target: targetCommit,
```

Comment:
Could we use the `shortenSHA` helper method so that when it is briefly displayed in the toolbar it isn't the long sha which is hard to digest/doesn't fit well in the toolbar. Also would apply in the `app-store` when `refreshAfterCheckout` is called.

Reply:
> Could we use the shortenSHA helper method 

Done.
![image](https://github.com/desktop/desktop/assets/90329875/0f4dae58-fa64-444e-90ed-3232811e2b02)

> Also would apply in the app-store when refreshAfterCheckout is called.

How should we figure out whether the `commitish` is a SHA or not?
We had changed to a commitish in 10f673f39260178ce94dcfccf2ee86bdb8d61233.

```ts
private async refreshAfterCheckout(
    repository: Repository,
    commitish: string
  ) {
    this.updateCheckoutProgress(repository, {
      kind: 'checkout',
      title: `Refreshing ${__DARWIN__ ? 'Repository' : 'repository'}`,
      description: 'Checking out',
      value: 1,
      target: commitish,
    })

    await this._refreshRepository(repository)
    return repository
  }
```

As a side-effect of the above, both commits and branches must now receive the same treatment.

And yes, the long SHA **is** _ugly_.

![image](https://github.com/desktop/desktop/assets/90329875/4d69b680-2404-41a9-a46b-d3b710517e82)

Reply:
We could pass in progress target as parameter, if called from checking out branch, progressTarget = branch name and if called from checkout out a commit, progressTarget = shortenSha(commit.sha)

Reply:
I handled it by shortening the SHA when it was passed as commitish.

Reply:
This was probably what you initially meant.